### PR TITLE
Improve Verbose message if content length is unknown in WebCmdlets

### LINKED ISSUE #19230: Regression from 7.3.3: Verbose Invoke-WebRequest / Invoke-WebRequest output is missing the byte count
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

```powershell
Invoke-RestMethod -Method HEAD http://example.org -Verbose
```

### Expected behavior

```console
VERBOSE: HTTP/1.1 HEAD with 0-byte payload
VERBOSE: received 648-byte response of content type text/html
```

### Actual behavior

```console
VERBOSE: HTTP/1.1 HEAD with 0-byte payload
VERBOSE: received -byte response of content type text/html
```

Note the missing number before "-byte".

### Error details

_No response_

### Environment data

```powershell
PowerShell Core 7.4.0-preview.1
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Show the verbose comment only if `Content-Length` is available.

## PR Context
Fixes #19230

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -551,11 +551,12 @@ protected override void ProcessRecord()
 
                             using HttpResponseMessage response = GetResponse(client, request, handleRedirect);
 
+                            string byteLength = response.Content.Headers.ContentLength is null ? string.Empty : $"{response.Content.Headers.ContentLength}-byte ";
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: Please use two resource string ("byte" should be localized).

**Comment by CarloToso** [comment]
Something like this @mklement0 ?

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 2]
```diff
@@ -552,12 +552,11 @@ protected override void ProcessRecord()
...
 
                             string contentType = ContentHelper.GetContentType(response);
-                            string respVerboseMsg = string.Format(
-                                CultureInfo.CurrentCulture,
-                                WebCmdletStrings.WebResponseVerboseMsg,
-                                response.Content.Headers.ContentLength,
-                                contentType);
-
+                            long? byteLength = response.Content.Headers.ContentLength ?? null;
+                            string respVerboseMsg = byteLength is null ?
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:556]:
```suggestion
                            string respVerboseMsg = response.Content.Headers.ContentLengthis null ?
```

Comment (mklement0) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:556]: LGTM, thanks. (@iSazonov, a space is missing before `is`).

**Comment by iSazonov** [comment]
@CarloToso Please fix build.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 3]
```diff
@@ -552,12 +552,10 @@ protected override void ProcessRecord()
...
-                            string respVerboseMsg = string.Format(
-                                CultureInfo.CurrentCulture,
-                                WebCmdletStrings.WebResponseVerboseMsg,
-                                response.Content.Headers.ContentLength,
-                                contentType);
-
+                            string respVerboseMsg = response.Content.Headers.ContentLength is null ?
+                                                    string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseVerboseMsgNoSize, contentType) :
+                                                    string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseVerboseMsg, response.Content.Headers.ContentLength, contentType);
+
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:558]:
```suggestion
                            long? contentLength = response.Content.Headers.ContentLength;
                            string respVerboseMsg = contentLength is null
                                ? string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseVerboseMsgNoSize, contentType)
                                : string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebResponseVerboseMsg, contentLength, contentType);
                            
```

**Review by daxian-dbw — APPROVED** [review]
LGTM with a couple suggestions.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs (hunk 3 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:558]: It's similar to my implementation a couple of commits ago, I will follow your suggestion later

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**